### DISCUSSION TITLE
rspamd:  failed to decode JSON response

### DISCUSSION THREAD

**Discussion by ghost** [rspamd/rspamd#4313] [Q&A] [answered]
Hello. I use OpenSMTPD as my MTA, and I recently started noticing that messages from other servers are not being sent to my mailbox as they should be. At the root of it, I found the message:

```
rspamd: failed to decode JSON response
```

In my `/var/log/maillog`. I have not modified the rspamd configuration at all except for adding a DKIM signing key, and things were working fine for the first few weeks until this happened recently.

```
mao$ tail /var/log/maillog
Oct 18 23:14:12 mao smtpd[71439]: senderscore: link-connect addr=[REDACTED] score=99
Oct 18 23:14:13 mao smtpd[92672]: 73fd66af244b6527 smtp tls ciphers=TLSv1.3:TLS_AES_256_GCM_SHA384:256
Oct 18 23:14:14 mao smtpd[71439]: rspamd: failed to decode JSON response
Oct 18 23:14:14 mao smtpd[92672]: 73fd66af244b6527 smtp failed-command command="DATA" result="421 server internal error"
Oct 18 23:14:14 mao smtpd[92672]: 73fd66af244b6527 smtp disconnected reason=quit
Oct 18 23:14:15 mao smtpd[92672]: 73fd66b0bd857727 smtp connected address=[REDACTED] [REDACTED]
Oct 18 23:14:15 mao smtpd[92672]: 73fd66b0bd857727 smtp tls ciphers=TLSv1.3:TLS_AES_256_GCM_SHA384:256
Oct 18 23:14:16 mao smtpd[71439]: rspamd: failed to decode JSON response
Oct 18 23:14:16 mao smtpd[92672]: 73fd66b0bd857727 smtp failed-command command="DATA" result="421 server internal error"
Oct 18 23:14:16 mao smtpd[92672]: 73fd66b0bd857727 smtp disconnected reason=quit
```

I am quite a noob with the sysadmin stuff, and did not expect this to suddenly happen after 2 or so weeks of it working normally.

If you are to ask for configuration, I will admit that I `s//g`'d from [Gilles' guide on setting up a OpenSMTPd server](https://poolp.org/posts/2019-09-14/setting-up-a-mail-server-with-opensmtpd-dovecot-and-rspamd/).

Thanks in advance.

**Comment by fatalbanana** [comment, marked as answer]
The problem is about changes to milter headers protocol response.

[filter-rspamd issue](https://github.com/poolpOrg/filter-rspamd/issues/41)
